reana_job_controller/kubernetes_job_manager.py

```diff
@@ -291,3 +296,10 @@ def _add_krb5_init_container(self, secrets_volume_mount):
                            )})
         self.job['spec']['template']['spec']['initContainers'].append(
             krb5_container)
+
+    def set_user_id(self, kubernetes_uid):
+        """Set user id for job pods. UIDs < 100 are refused for security."""
+        if kubernetes_uid >= 100:
```

Comment:
Note we need to check `kubernetes_uid` presence and type, something like:

```python
if kubernetes_uid and int(kubernetes_uid) >= 100
```

and also catch any `ValueError` and friends...

Reply:
Somebody is passing `None`, the default value is not enough:

```
reana ❯ kubectl logs batch-yadage-3aeb184e-a182-4b23-baca-7ece735bdf45-2cxbz job-controller
 * Serving Flask app "reana_job_controller/app.py"
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
/usr/local/lib/python3.6/site-packages/htcondor/__init__.py:25: UserWarning:
Using a null condor_config.
Neither the environment variable CONDOR_CONFIG, /etc/condor/,
/usr/local/etc/, nor ~/condor/ contain a condor_config source.
  warnings.warn(message)
2019-12-13 14:09:41,824 | werkzeug | MainThread | INFO |  * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
2019-12-13 14:09:50,271 | werkzeug | Thread-1 | INFO | 127.0.0.1 - - [13/Dec/2019 14:09:50] "GET /jobs HTTP/1.1" 200 -
2019-12-13 14:09:56,347 | reana_job_controller.factory | Thread-2 | ERROR | Exception on /jobs [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/reana_job_controller/rest.py", line 206, in create_job
    **job_request)
  File "/usr/local/lib/python3.6/site-packages/reana_job_controller/kubernetes_job_manager.py", line 84, in __init__
    self.set_user_id(kubernetes_uid)
  File "/usr/local/lib/python3.6/site-packages/reana_job_controller/kubernetes_job_manager.py", line 302, in set_user_id
    if kubernetes_uid >= 100:
TypeError: '>=' not supported between instances of 'NoneType' and 'int'
2019-12-13 14:09:56,350 | werkzeug | Thread-2 | INFO | 127.0.0.1 - - [13/Dec/2019 14:09:56] "POST /jobs HTTP/1.1" 500 -
```